agent/balrogagent/cmd.py

```diff
@@ -0,0 +1,70 @@
+import aiohttp
+import asyncio
+import logging
+import time
+import traceback
+
+from . import client
+from .changes import get_telemetry_uptake, is_ready
+from .log import configure_logging
+
+
+async def run_agent(loop, balrog_api_root, balrog_username, balrog_password, telemetry_api_root, sleeptime=30):
+    auth = aiohttp.BasicAuth(balrog_username, balrog_password)
+
+    while True:
+        try:
+            async with aiohttp.ClientSession(auth=auth, loop=loop) as session:
+                logging.debug("Looking for active scheduled changes...")
+                # TODO: sometimes getting "unclosed response" and "unclosed connection" errors.
+                resp = await client.request(session, balrog_api_root, "/csrf_token", method="HEAD")
+                csrf_token = resp.headers["X-CSRF-Token"]
+                resp = await client.request(session, balrog_api_root, "/scheduled_changes/rules")
+                sc = (await resp.json())["scheduled_changes"]
+                logging.debug("Found %s", len(sc))
+                for change in sc:
+                    logging.debug("Processing change %s", change["sc_id"])
+                    ready_kwargs = {}
+                    if change["telemetry_uptake"]:
```

Comment:
IIRC my thought at the time was that here we know those keys exist, and in changes.py we don't. In retrospect, I may have been influenced by the fact that it was simpler to write the tests for changes.py by using .get().